### DISCUSSION TITLE
Revalidate request in the background, after returning the response

### DISCUSSION THREAD

**Discussion by mhsfh** [arthurfiorette/axios-cache-interceptor#640] [Q&A] [answered]
Hello,
first thanks for this great tool,

I need to return old (stale) data and revalidate it in the background after I return the data.
how can I achieve this?

thanks

**Comment by arthurfiorette** [comment]
Please use hydrate: https://axios-cache-interceptor.js.org/config/request-specifics#cache-hydrate

Else, you can just create another call to axios.get() without await, which will revalidate in the background. Automatic support for this feature is planned to be implemented but have some architectural problems, as we cannot resolve a request and still call its adapters.

**Reply by mhsfh** [reply]
thanks for your quick reply,
hydrate is not suitable for my need as I'm using the package in the backend side

I will try to use the second solution, thanks

**Reply by arthurfiorette** [reply]
Hydrate does not have anything to do with frontends. Just its main usecase is for it. You can just use hydrate to return a callback with current data and return the promise with the requested one.

**Reply by mhsfh** [reply]
@arthurfiorette 
i really appreciate your help,

i'm using `axios-cache-interceptor`  as follows in nest.js app

```ts
export const AXIOS_INSTANCE = Axios.create({
  baseURL: 'www.example.com',
});

AXIOS_INSTANCE.defaults.headers.common = {
  Accept: 'application/json',
  'Content-Type': 'application/json',
};
const axiosCacheInstance = setupCache(AXIOS_INSTANCE, {
  debug: logger.log,
  storage: redisStorage,
  methods: ['get', 'post'],
  ttl: 1000,
  headerInterpreter: () => 'not enough headers',
  hydrate: async (cache) => {
    console.log('this is the preivous data', cache);
  },
});

export const customInstance = <T>(
  config: AxiosRequestConfig,
  options?: CacheRequestConfig,
) => {
  return axiosCacheInstance<T>({ ...config, ...options }) 
};
```

but hydrate doesn't log anything, is there something I'm doing it wrongly

**Comment by arthurfiorette** [comment, marked as answer]
For future ref, this twitter thread my help:

https://twitter.com/adamdehaven/status/1733228594313277608
